### PR TITLE
Sedona 173 delta lake tests

### PR DESCRIPTION
**Pull request by Imbruced**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the assoicated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-173. The PR name follows the format `[SEDONA-173] my subject`.

## What changes were proposed in this PR?
delta lakes tests in scala and python notebook with examples how to use geospatial functions with delta lake.

## How was this patch tested?
Unit tests

### CODE REVIEW

File: delta/pom.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            <groupId>org.apache.sedona</groupId>
+            <artifactId>sedona-core-${spark.compat.version}_${scala.compat.version}</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.delta</groupId>
+            <artifactId>delta-core_2.12</artifactId>
+            <version>2.0.0</version>
+        </dependency>
```
Comment (jiayuasu) [inline delta/pom.xml:36]: The scope should be `test`

File: delta/pom.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <artifactId>sedona-parent</artifactId>
+        <groupId>org.apache.sedona</groupId>
+        <version>1.1.1-incubating</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>delta</artifactId>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
```
Comment (jiayuasu) [inline delta/pom.xml:15]: Please add `<maven.deploy.skip>false</maven.deploy.skip>` . This will prevent this module from being published to Maven Central.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Comment by Kimahriman** [comment]
Just my opinion, this seems like the only thing this is really "testing" is if Delta supports user-defined types, which is more of a thing for the Delta repo than this one, and doesn't really seem like it makes sense as a new module to add. But this makes perfect sense to just be an example notebook on how you can work with geometries natively with Delta.

**Comment by Imbruced** [comment]
@Kimahriman I also agree, the main reason why I created additional module was to figure out later if we can add by default writing to delta in geoparquet data format.

**Comment by jiayuasu** [comment]
@Imbruced Yes, I agree with @Kimahriman . The only purpose of this module is testing. Can you (1) place the Scala/Java part to /example/ folder (2) keep the Python notebook in /binder/  ?

**Pull request closed without merging**